pkg/api/vlabs/validate.go

```diff
@@ -523,3 +552,17 @@ func GetVNETSubnetIDComponents(vnetSubnetID string) (string, string, string, str
 	}
 	return submatches[1], submatches[2], submatches[3], submatches[4], nil
 }
+
+// ip4BroadcastAddress returns the broadcast address for the given IP subnet.
+func ip4BroadcastAddress(n *net.IPNet) net.IP {
+	ip4 := n.IP.To4()
```

Comment:
Please add some comments around what you are doing in these blocks. I can see them becoming confusing down the line